include/sys/arch_interface.h

```diff
@@ -645,6 +645,24 @@ FUNC_NORETURN void arch_syscall_oops(void *ssf);
 size_t arch_user_string_nlen(const char *s, size_t maxsize, int *err);
 #endif /* CONFIG_USERSPACE */
 
+#ifdef KERNEL_COHERENCE
+/**
+ * @brief Detect memory coherence type
+ *
+ * Required when ARCH_HAS_COHERENCE is true.  This function returns
+ * true if the byte pointed to lies within an architecture-defined
+ * "coherence region" (typically implemented with uncached memory) and
+ * can safely be used in multiprocessor code without explicit flush or
+ * invalidate operations.
+ *
+ * @note The result is for only the single byte at the specified
+ * address, this API is not required to check region boundaries or to
+ * expect misaligned pointers.  The expectation is that the code above
+ * will have queried the appropriate address(es).
+ */
+bool arch_mem_coherent(void *ptr);
```

Comment:
Could we make this static inline ?

Reply:
It actually is.  This is the API declaration for our official architecture API, mostly for the benefit of documentation clarity.  The implementation is in the arch-provided kernel_arch_func.h (though see note there: REALLY this is coming out of the SoC layer, not the xtensa arch, but we don't have a framework for SoC headers).